src/components/utils/__tests__/onfidoApi/sendMultiframeSelfie.test.ts

```diff
@@ -47,105 +62,119 @@ describe('onfidoApi', () => {
-          status: 401,
-          response: { error: 'unauthorized' },
+      it('should call onError callback', () => {
+        return new Promise((okay, error) => {
+          const mockXHR = createMockXHR({
+            status: 401,
+            response: { error: 'unauthorized' },
+          })
+          jest.spyOn(window, 'XMLHttpRequest').mockImplementation(() => mockXHR)
+
+          sendMultiframeSelfie(
+            snapshotData,
+            selfieData,
+            jwtToken,
+            url,
+            () => {
+              error('success should not be called')
+            },
+            (e: ParsedError) => {
+              try {
+                expect(e.status).toEqual(401)
+                expect(e.response).toEqual({ error: 'unauthorized' })
+                okay(null)
+              } catch (e) {
+                error(e)
+              }
+            },
+            mockedTrackingCallback
+          )
+
+          mockXHR.onload && mockXHR.onload(new ProgressEvent('error'))
+
+          expect(mockXHR.send).toHaveBeenCalledTimes(1)
         })
       })
     })
 
     describe('with invalid data', () => {
-      beforeEach(() => {
-        mockXHR = createMockXHR({})
-        jest.spyOn(window, 'XMLHttpRequest').mockImplementation(() => mockXHR)
-      })
-
       it('should call onError callback with TypeError', async () => {
-        const invalidSnapshotData = { ...snapshotData, blob: {} as Blob }
-        const invalidSelfieData = { ...selfieData, blob: {} as Blob }
-
-        sendMultiframeSelfie(
-          invalidSnapshotData,
-          invalidSelfieData,
-          jwtToken,
-          url,
-          mockedOnSuccess,
-          mockedOnError,
-          mockedTrackingCallback
-        )
-
-        mockXHR &&
-          mockXHR.onload &&
-          mockXHR.onload(new ProgressEvent('Type error'))
-        await runAllPromises()
-
-        expect(mockXHR.send).not.toHaveBeenCalled()
-        expect(mockedOnSuccess).not.toHaveBeenCalled()
-        expect(mockedOnError).toHaveBeenCalledTimes(1)
-
-        const error = mockedOnError.mock.calls[0][0]
-        expect(error).toMatchObject(/TypeError/)
-        expect(error.message).toEqual(
-          `Failed to execute 'append' on 'FormData': parameter 2 is not of type 'Blob'.`
-        )
+        return new Promise((okay, error) => {
+          xhrMock = createMockXHR({})
+          jest.spyOn(window, 'XMLHttpRequest').mockImplementation(() => xhrMock)
+
+          const invalidSnapshotData = { ...snapshotData, blob: {} as Blob }
+          const invalidSelfieData = { ...selfieData, blob: {} as Blob }
+
+          sendMultiframeSelfie(
+            invalidSnapshotData,
+            invalidSelfieData,
+            jwtToken,
+            url,
+            () => {
+              error('success should not be invoked')
+            },
+            (e) => {
+              // @ts-ignore
+              const ex: TypeError = e as TypeError
```

Comment:
Can we call this `error` or `err`? `ex` is not as readable

Reply:
Naming ... 

`error` is used already.
`ex` is for exception

Anyway, renamed it for err. Feel free in the future to rename it by yourself, to save both of us time.